background/load-addon-manifests.js

```diff
@@ -21,6 +21,11 @@
           info.text = scratchAddons.l10n.get(`${folderName}/@info-${info.id}`, {}, info.text);
         }
       }
+      if (manifest.credits) {
+        for (const credit of manifest.credits || []) {
+          credit.note = scratchAddons.l10n.get(`${folderName}/@credits-${credit.id}`, {}, credit.note);
```

Comment:
This does not guarantee credit.id is set. If it is not set, credit.note ends up addonName/@credits-undefined.

Reply:
@RedGuy12 Such errors should be caught by the schema validator.

Reply:
either both `id` and `note` is set, or neither is set. if both are set, credit.note gets reassigned to the message `addonName/@credits-creditId`. If neither are set, **this line still runs**. We forgot to add any conditional here. it tries to fetch the string with id `addonName/@credits-undefined`.  No string is found so it falls back to the key, `"addonName/@credits-undefined"`.